source/server/session_repository.cpp

```diff
@@ -27,12 +34,24 @@ int SessionRepository::add_session(
     const std::string& session_name,
     InitFunc init_func,
     CleanupSessionFunc cleanup_func,
-    uint32_t& session_id)
+    uint32_t& session_id,
+    SessionInitializationBehavior initializationBehavior,
+    bool* createdNewSession)
```

Comment:
I like the term "create", but I hesitate to introduce it when it comes to sessions.

IVI (or at least NI's use of IVI) has the term "initialize" to mean a session was created. Never do the APIs say "create" nor "open".

That's why I think it's useful to stick with two verbes: "attach" and "initialize" rather than overload "initialize" to mean "sometimes create sometimes attach".

Reply:
I am aware that this applies to our MI and RF APIs but DAQmx uses different terminology and I bet other drivers do as well.

Reply:
Here are all the methods that call the session manager's `add_session` method:
```
nidaqmx: CreateTask, CreateWatchdogTimer, CreateWatchdogTimerTask/Ex, LoadTask
nidcpower: InitializeWithChannels, InitializeWithIndependentChannels
nidigitalpattern: Init, InitWithOptions
nidmm: Init, InitWithOptions
nifgen: Init, InitWithOptions, InitializeWithChannels
nirfmxbluetooth, nirfmxinstr, nirfmxlte, nirfmxnr, nirfmxspecan, nirfmxwlan: Initialize, InitializeFromNIRFSASession
nirfsa: Init, InitWithOptions
nirfsg: Init, InitWithOptions
niscope: Init, InitWithOptions
niswitch: Init, InitWithOptions, InitWithTopology
nisync: Init, InitExtCal
nitclk: N/A
nixnet: CreateSession, CreateSessionByRef, DbOpenDatabase, SystemOpen
nixnetsocket: Accept, IpStackCreate, IpStackOpen, Socket
```

Reply:
I've changed it to consistently use "initialize" now since we have to pick one, and that term is used more often than not, at least in what we support in gRPC-device today.